doc/neps/nep-0049.rst

```diff
@@ -0,0 +1,369 @@
+        if (p != NULL) {
+            char *real = (char *)p - 64;
+            if (strncmp(real, "originally allocated", 20) != 0) {
+                fprintf(stdout, "uh-oh, unmatched shift_realloc\\n");
+                return realloc(p, sz);
+            }
+            return (void *)((char *)realloc(real, sz + 64) + 64);
+        }
+        else {
+            char *real = (char *)realloc(p, sz + 64);
+            if (real == NULL) {
+                return NULL;
+            }
+            snprintf(real, 64, "originally allocated "
+                     "%ld  via realloc", (unsigned long)sz);
+            return (void *)(real + 64);
+        }
+    }
+
+    static PyDataMem_Handler new_handler = {
+        "secret_data_allocator",
+        shift_alloc,      /* alloc */
+        shift_zero,       /* zeroed_alloc */
+        shift_free,       /* free */
+        shift_realloc     /* realloc */
+    };
+
+    static PyObject* mem_policy_test_prefix(PyObject *self, PyObject *args)
+    {
+    
+        if (!PyArray_Check(args)) {
+            PyErr_SetString(PyExc_ValueError,
+                    "must be called with a numpy scalar or ndarray");
+        }
+        return PyUnicode_FromString(
+                        PyDataMem_GetHandlerName((PyArrayObject*)args));
+    };
+
+    static PyObject* mem_policy_set_new_policy(PyObject *self, PyObject *args)
+    {
+    
+         const PyDataMem_Handler *old = PyDataMem_SetHandler(&new_handler);
+         return PyUnicode_FromString(old->name);
+     
+    };
+    
+    static PyObject* mem_policy_set_old_policy(PyObject *self, PyObject *args)
+    {
+    
+         const PyDataMem_Handler *old = PyDataMem_SetHandler(NULL);
+         return PyUnicode_FromString(old->name);
+     
+    };
+        
+    static PyMethodDef methods[] = {
+    {"test_prefix", (PyCFunction)mem_policy_test_prefix, METH_O},
+    {"set_new_policy", (PyCFunction)mem_policy_set_new_policy, METH_NOARGS},
+    {"set_old_policy", (PyCFunction)mem_policy_set_old_policy, METH_NOARGS},
+    { NULL }
+    };
+
+    static struct PyModuleDef moduledef = {
+        PyModuleDef_HEAD_INIT,
+        "mem_policy",  /* m_name */
+        NULL,           /* m_doc */
+        -1,             /* m_size */
+        methods,        /* m_methods */
+    };
+
+    PyMODINIT_FUNC
+    PyInit_mem_policy(void) {
+    PyObject *mod = PyModule_Create(&moduledef);
+        import_array();
+        return mod;
+    }
+ 
+
+Related Work
+------------
+
+The NEP is being tracked by the pnumpy_ project and a `comment in the PR`_
+mentions use in orchestrating FPGA DMAs.
+
+Implementation
+--------------
+
+The NEP has been implemented in `PR  17582`_.
+
+Alternatives
+------------
+
+These were discussed in `issue 17467`_. `PR 5457`_  and `PR 5470`_ proposed a
+global interface for specifying aligned allocations. 
+
+``PyArray_malloc_aligned`` and friends were added to NumPy with the random API
+refactor. and are used there for performance.
```

Comment:
Question: is there a link to `PyArray_malloc_aligned`, or is it internal?
```suggestion
``PyArray_malloc_aligned`` and friends were added to NumPy with the ``random`` API
refactor and are used there for performance.
```

Reply:
> Question: is there a link to PyArray_malloc_aligned, or is it internal?

It is internal